src/sentry/preprod/api/endpoints/project_preprod_artifact_size_analysis_download.py

```diff
@@ -0,0 +1,94 @@
+import posixpath
+
+from django.conf import settings
+from django.http.response import FileResponse, HttpResponseBase
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import analytics, features
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.models.files.file import File
+from sentry.preprod.models import PreprodArtifact, PreprodArtifactSizeMetrics
+
+
+@region_silo_endpoint
+class ProjectPreprodArtifactSizeAnalysisDownloadEndpoint(ProjectEndpoint):
+    owner = ApiOwner.EMERGE_TOOLS
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+
+    def get(self, request: Request, project, artifact_id) -> HttpResponseBase:
+        """
+        Download size analysis results for a preprod artifact
+        ````````````````````````````````````````````````````
+
+        Download the size analysis results for a preprod artifact.
+
+        :pparam string organization_id_or_slug: the id or slug of the organization the
+                                          artifact belongs to.
+        :pparam string project_id_or_slug: the id or slug of the project to retrieve the
+                                     artifact from.
+        :pparam string artifact_id: the ID of the preprod artifact to download size analysis for.
+        :auth: required
+        """
+
+        analytics.record(
+            "preprod_artifact.api.size_analysis_download",
+            organization_id=project.organization_id,
+            project_id=project.id,
+            user_id=request.user.id,
+            artifact_id=artifact_id,
+        )
+
+        if not settings.IS_DEV and not features.has(
```

Comment:
lets just reuse the same ff (apologies, i could have made the naming more generic but i dont think its a big deal)